app/grandchallenge/retina_api/views.py

```diff
@@ -896,3 +905,137 @@ class RetinaImagePathologyAnnotationViewSet(viewsets.ModelViewSet):
+                    "name": patient.name,
+                    "subfolders": sorted(
+                        generate_studies(patient.study_set),
+                        key=lambda x: x["id"],
+                    ),
+                    "images": [],
+                }
+
+        def generate_studies(study_list):
+            for study in study_list.all():
+                yield {
+                    "id": study.id,
+                    "name": study.name,
+                    "images": sorted(
+                        generate_images(study.image_set.all()),
+                        key=lambda x: x["id"],
+                    ),
+                    "subfolders": [],
+                }
+
+        def generate_images(image_list):
+            for image in image_list:
+                yield {
+                    "id": image.id,
+                    "name": image.name,
+                    "eye": image.eye_choice,
+                    "modality": image.modality.modality,
+                    "patient": image.study.patient.name,
+                    "study": image.study.name,
+                    "archives": image.archive_set.values(),
+                }
+
+        return sorted(
+            generate_archives(archives, patients), key=lambda x: x["id"]
+        )
+
+    def get(self, request):
+        archives = Archive.objects.all().prefetch_related(
+            "images", "images__study"
+        )
+        patients = Patient.objects.all().prefetch_related(
+            "study_set",
+            "study_set__image_set",
+            "study_set__image_set__modality",
+            "study_set__image_set__obs_image",
+            "study_set__image_set__oct_image",
+            "study_set__image_set__archive_set",
+        )
+        return Response(self.create_response_object(archives, patients))
+
+
+class B64ThumbnailAPIView(RetrieveAPIView):
+    permission_classes = (RetinaAPIPermission,)
+    authentication_classes = (authentication.TokenAuthentication,)
+
+    @staticmethod
+    def get_image_itk(image_id, user):
+        image_object = get_object_or_404(Image, pk=image_id)
+
+        if not user_can_download_image(user=user, image=image_object):
+            raise PermissionDenied
+
+        image_itk = image_object.get_sitk_image()
+        if image_itk is None:
+            raise Http404
+
+        return image_itk
+
+    @staticmethod
+    def convert_itk_to_pil(image_itk):
+        depth = image_itk.GetDepth()
+        image_nparray = sitk.GetArrayFromImage(image_itk)
+        if depth > 0:
+            # Get center slice of image if 3D
+            image_nparray = image_nparray[depth // 2]
+        return PILImage.fromarray(image_nparray)
+
+    @staticmethod
+    def create_thumbnail_as_base64(image_pil, width, height):
+        image_pil.thumbnail((width, height), PILImage.ANTIALIAS)
+        buffer = BytesIO()
+        image_pil.save(buffer, format="png")
+        return base64.b64encode(buffer.getvalue())
+
+    def get(
```

Comment:
I think caching can be useful here. The performance is not great since the object is first opened as SimpleITK image, then converted to Pillow image, then resized and saved into a bytes buffer and encoded to a Base64 string. I implemented caching in Django by using `cache_page` in the urls.py file.